.github/workflows/windows.yml

```diff
@@ -77,36 +77,75 @@ jobs:
         echo "MKLROOT=${{ github.workspace }}\mkl" | Out-File -FilePath $env:GITHUB_ENV  -Encoding utf8 -Append
       shell: powershell
 
-    - name: Prepare vcpkg
-      uses: lukka/run-vcpkg@v7.4
-      with:
-        vcpkgArguments: protobuf pcre2 --overlay-ports="${{ github.workspace }}\vcpkg-override\ports\pcre2"
-        vcpkgGitCommitId: 8dddc6c899ce6fdbeab38b525a31e7f23cb2d5bb
-        vcpkgDirectory: ${{ github.workspace }}/vcpkg/
-        vcpkgTriplet: x64-windows-static
-
-    # Windows CPU only minimal build
-    - name: Build Release # @TODO this is actually a debug build until the ninja generator gets fixed
-      uses: lukka/run-cmake@v3
+    - name: Cache vcpkg
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-vcpkg
       with:
-        buildDirectory: ${{ github.workspace }}/build
-        cmakeAppendedArgs: '-G Ninja
-          -DCMAKE_BUILD_TYPE="Release"
-          -DUSE_WASM_COMPATIBLE_SOURCE="OFF"
-          -DUSE_STATIC_LIBS="TRUE" 
-          -DCMAKE_CXX_COMPILER_LAUNCHER=${{github.workspace}}\ccache.exe
-          -DCMAKE_C_COMPILER_LAUNCHER=${{github.workspace}}\ccache.exe
-        '
-        cmakeListsOrSettingsJson: CMakeListsTxtAdvanced
-        cmakeListsTxtPath: ${{ github.workspace }}/CMakeLists.txt
-        useVcpkgToolchainFile: true
-        cmakeBuildType: Release
+
+        path: C:\vcpkg\installed # cache only the latest installed packages
+        key: ${{ runner.os }}-build-${{ env.cache-name }}
+        restore-keys: |-
+            ${{ runner.os }}-build-${{ env.cache-name }}
+            ${{ runner.os }}-build
+            
+    - name: Disable debug vcpkg build
+      shell: powershell
+      working-directory: C:\vcpkg\triplets
+      run: |
+         $PSDefaultParameterValues['Out-File:Encoding'] = 'utf8' # Powershell murders me.
+         echo "set(VCPKG_BUILD_TYPE release)" | Tee-Object -FilePath x64-windows-static.cmake -Append
+         echo "set(VCPKG_BUILD_TYPE release)" | Tee-Object -FilePath x64-windows.cmake -Append
+         cat x64-windows-static.cmake
+         cat x64-windows.cmake
+
+    - name: Install dependencies with vcpkg
+      working-directory: C:\vcpkg
+      run: |
+        $Env:VCPKG_BUILD_TYPE = 'release'
+        $Env:VCPKG_DEFAULT_TRIPLET = 'x64-windows-static' # QT6 version, linguist tools not working yet: qtbase:x64-windows-static qttools:x64-windows-static qtsvg:x64-windows-static
+        .\vcpkg install protobuf:x64-windows-static pcre2:x64-windows-static 
+        .\vcpkg upgrade --no-dry-run # In case there are new builds available after cache restoration
+      shell: powershell
+        
+    - name: Fix PCRE2 extensions # https://github.com/ugermann/ssplit-cpp/pull/29
```

Comment:
This may not be necessary anymore after the merge of my "fix" but I haven't tested it.